elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexTransaction.java

```diff
@@ -241,14 +241,12 @@ public void setAttribute(Object entity, String attributeName, Object attributeVa
     private Serializable extractId(FilterExpression filterExpression,
                                    String idFieldName,
                                    Object parent,
-                                   Class<?> relationClass,
                                    EntityDictionary dictionary) {
         Collection<FilterPredicate> predicates = filterExpression.accept(new PredicateExtractionVisitor());
         for (FilterPredicate predicate : predicates) {
             List<Object> values = predicate.getValues();
             Class<?> entityClass = dictionary.getParameterizedType(parent, predicate.getField());
-            if (relationClass == entityClass
```

Comment:
Removing this makes the logic incorrect IMO.  What if the client had a filter predicate on two different entity types (that also had the same ID field name).  We might grab the wrong ID here.